src/Connection/ConnectionInterface.php

```diff
@@ -19,9 +20,9 @@ public function connect(): Connection;
 
     /**
      * @param MessageInterface $message
-     * @return array
+     * @return ResponseInterface
      */
-    public function run(MessageInterface $message): array;
+    public function run(MessageInterface $message);
```

Comment:
Add strict return type ResponseInterface

Reply:
The ResponseInterface will be wrapped by a Cursor later.